fontforge/woff.c

```diff
@@ -348,11 +348,49 @@ return( NULL );
     }
 
     if ( sf!=NULL && metaOffset!=0 ) {
+	/*
+	* Boundary/integer overflow checks:
+	*
+	* We don't want to actually dereference a null pointer (returned
+	* by asking to allocate too much RAM) and we don't want to allocate
+	* a 0-sized chunk (caused when one of the (metaLenxxx + 1) values overflows).
+	*
+	* uncompress() will always safely return an error if
+	* metaLenCompressed > metaLenUncompressed, so check #1 prevents
+	* this. This check, in conjunction with #2, also prevents calling
+	* malloc(0) because 0xFFFFFFFF is the largest possible value for 
+	* an unsigned 32bit int and the case where both values are 0xFFFFFFFF 
+	* is prevented by check #2.
+	*
+	* We can safely pass sf->woffMetadata as a NULL pointer because
+	* it's never accessed anywhere else without a check for it being
+	* NULL first
+	*/
+	if(metaLenCompressed > metaLenUncompressed) { // Check #1 to prevent uncompress from returning an error
```

Comment:
Oops, I misinterpreted what the uncompress() function does. I had it in my head that the length of the source being longer than the length of the destination buffer would always cause uncompress() to return Z_BUF_ERROR but that doesn't appear to be the case.